sycl/doc/PreprocessorMacros.md

```diff
@@ -55,24 +55,7 @@ This file describes macros that have effect on SYCL compiler and run-time.
 - **SYCL2020_CONFORMANT_APIS**
   This macro is used to comply with the SYCL 2020 specification, as some of the current 
   implementations may be widespread and not conform to it.
-  Description of what it changes:
-  1) According to spec, `backend_return_t` for opencl event 
-  should be `std::vector<cl_event>` instead of `cl_event`. Defining this macro 
-  will change the behavior of `sycl::get_native()` function and using types for 
-  next structs: `interop<backend::opencl, event>`, `BackendInput<backend::opencl, event>`, 
-  `BackendReturn<backend::opencl, event>` to be in line with the spec.
-  2) According to spec, `backend_return_t` for opencl buffer 
-  should be `std::vector<cl_mem>` instead of `cl_mem`. Defining this macro 
-  will change the behavior of `interop_handle::get_native_mem()` and `sycl::get_native()` functions
-  and using type for `BackendReturn<backend::opencl, buffer>` to be in line with the spec.
-  3) According to spec, `sycl::buffer_allocator` should be a template class taking a single
-  type parameter denoting the data type of the associated buffer. Likewise, `sycl::buffer`
-  with that take an allocator as a constructor argument should use
-  `sycl::buffer_allocator<std::remove_const_t<T>>`  by default, where `T` is the data type of
-  that buffer. Defining this macro will change the definition of `sycl::buffer_allocator` to
-  be templated and `sycl::buffer` will be using `sycl::buffer_allocator<std::remove_const_t<T>>`
-  by default, where `T` is the data type of that buffer, if it is not explicitly given an
-  allocator.
+  Defining this macro currently has no effect on the API.
```

Comment:
Should we remove it completely then? Maybe (hopefully) next use will be SYCL202Y and not 2020...

Reply:
The motivation for keeping it is so we can reuse it for any SYCL 2020 changes we may have missed that break API/ABI after we once again close the breakage window. We could always reintroduce it of course, but this way we should hopefully keep the same name for the macro as before if we run into a case like that.

The primary benefit is that users that do not worry about ABI and just want SYCL 2020 features as soon as they are ready can just keep defining this in their project and they will get it automatically as we add such missing features.

@pvchupin - What are your thoughts on this?

Reply:
I think the change proposed is the best at this point.
It would be great to track carefully documentation and implementation. It would be great to do everything in this window. But in reality I guess we will miss something or find problem late. So keeping macro around should be fine, as a good predefined/agreed name.